index.html

```diff
@@ -3867,9 +3906,147 @@ <h3>Parsing content</h3>
+    <ol class=algorithm data-link-for="NDEFRecord">
+      <li>
+        Set |record|'s <a>recordType</a> to "`:s`".
+      </li>
+      <li>
+        Set |record|'s <a>mediaType</a> to `null`.
+      </li>
+      <li>
+        If |ndefRecord|'s <a>PAYLOAD field</a> has not exactly 4 bytes,
+        [= exception/throw =] a {{TypeError}} and abort these steps.
+      </li>
+      <li>
+        Let |buffer:byte sequence| be the <a>byte sequence</a> of
+        |ndefRecords|'s <a>PAYLOAD field</a>.
+      </li>
+      <li>
+        Set |record|'s <a>data</a> to |buffer|.
+        <p class="note">
+          Applications can parse this value as a 32 bit unsigned integer that
+          denotes the size of the object the URI record in the <a>smart-poster</a>
+          refers to.
+        </p>
+      </li>
+      <li>
+        Return |record|.
+      </li>
+    </ol>
+  </div>
+  <div>
+    To <dfn>parse a smart-poster type record</dfn> given an |ndefRecord| into a
+    |record:NDEFRecord|, run these steps:
+    <ol class=algorithm data-link-for="NDEFRecord">
+      <li>
+        Set |record|'s <a>recordType</a> to "`:t`".
+      </li>
+      <li>
+        Set |record|'s <a>mediaType</a> to `null`.
+      </li>
+      <li>
+        Let |buffer:byte sequence| be the <a>byte sequence</a> of
+        |ndefRecords|'s <a>PAYLOAD field</a>.
+        <p class="note">
+          Applications can parse this value as a string that contains an
+          [[RFC2048]] media type that denotes the media type of the object the
+          URI record in the <a>smart-poster</a> refers to.
+        </p>
+      </li>
+      <li>
+        Set |record|'s <a>data</a> to |buffer|.
+      </li>
+      <li>
+        Return |record|.
+      </li>
+    </ol>
+  </div>
+  <div>
+    To <dfn>parse a smart-poster action record</dfn> given an |ndefRecord| into
+    a |record:NDEFRecord|, run these steps:
+    <ol class=algorithm data-link-for="NDEFRecord">
+      <li>
+        Set |record|'s <a>recordType</a> to "`:act`".
+      </li>
+      <li>
+        Set |record|'s <a>mediaType</a> to `null`.
+      </li>
+      <li>
+        If |ndefRecord|'s <a>PAYLOAD field</a> has not exactly 1 byte,
+        [= exception/throw =] a {{TypeError}} and abort these steps.
+      </li>
+      <li>
+        Let |buffer:byte sequence| be the <a>byte sequence</a> of
+        |ndefRecords|'s <a>PAYLOAD field</a>.
+      </li>
+      <li>
+        Set |record|'s <a>data</a> to |buffer|.
+        <p class="note">
+          Applications can parse this value as an 8 bit unsigned integer for
+          which the values are defined <a href="#dfn-action-record">here</a>.
+        </p>
+      </li>
+      <li>
+        Return |record|.
+      </li>
+    </ol>
+  </div>
+  <div>
+    To <dfn>validate smart poster icon type</dfn> given |type:string|,
+    run these steps:
+    <ol class=algorithm data-link-for="NDEFRecord">
+      <li>
+        If |type| starts with `"image"` or `"video"`, return `true`.
```

Comment:
Shall we add an extra `/` after `image` and `video`?

Reply:
Yes, fixed (amended).